libvcx/src/utils/provision.rs

```diff
@@ -6,312 +6,46 @@ use agency_client::agent_utils;
-                        my_vk: &str,
-                        agent_did: &str,
-                        agent_vk: &str,
-                        wallet_name: &str,
-                        my_config: &Config) -> VcxResult<String> {
-    let (issuer_did, issuer_vk) = _create_issuer_keys(my_did, my_vk, my_config)?;
-
-    let mut final_config = json!({
-        "wallet_key": &my_config.wallet_key,
-        "wallet_name": wallet_name,
-        "agency_endpoint": &my_config.agency_url,
-        "agency_did": &my_config.agency_did,
-        "agency_verkey": &my_config.agency_verkey,
-        "sdk_to_remote_did": my_did,
-        "sdk_to_remote_verkey": my_vk,
-        "institution_did": issuer_did,
-        "institution_verkey": issuer_vk,
-        "remote_to_sdk_did": agent_did,
-        "remote_to_sdk_verkey": agent_vk,
-        "institution_name": get_or_default(&my_config.name, "<CHANGE_ME>"),
-        "genesis_path": get_or_default(&my_config.path, "<CHANGE_ME>")
-    });
-
-    if let Some(key_derivation) = &my_config.wallet_key_derivation {
-        final_config["wallet_key_derivation"] = json!(key_derivation);
-    }
-    if let Some(wallet_type) = &my_config.wallet_type {
-        final_config["wallet_type"] = json!(wallet_type);
-    }
-    if let Some(_storage_config) = &my_config.storage_config {
-        final_config["storage_config"] = json!(_storage_config);
-    }
-    if let Some(_storage_credentials) = &my_config.storage_credentials {
-        final_config["storage_credentials"] = json!(_storage_credentials);
-    }
-    if let Some(_pool_config) = &my_config.pool_config {
-        final_config["pool_config"] = json!(_pool_config);
-    }
-    if let Some(_webhook_url) = &my_config.webhook_url {
-        final_config["webhook_url"] = json!(_webhook_url);
-    }
-    if let Some(_use_latest_protocols) = &my_config.use_latest_protocols {
-        final_config["use_latest_protocols"] = json!(_use_latest_protocols);
-    }
-
-    Ok(final_config.to_string())
-}
-
-pub fn connect_register_provision(config: &str) -> VcxResult<String> {
-    debug!("connect_register_provision >>> config: {:?}", config);
-    let my_config = parse_config(config)?;
-
-    trace!("***Configuring Library");
-    set_config_values(&my_config);
-
-    trace!("***Configuring Wallet");
-    let (my_did, my_vk, wallet_name, _) = configure_wallet(&my_config)?;
-
-    debug!("connect_register_provision:: Final settings: {:?}", settings::settings_as_string());
-
-    trace!("Connecting to Agency");
-    let (agent_did, agent_vk) = agent_utils::onboarding(&my_did, &my_vk, &my_config.agency_did)?;
-
-    let config = get_final_config(&my_did, &my_vk, &agent_did, &agent_vk, &wallet_name, &my_config)?;
-
-    wallet::close_main_wallet()?;
-
-    Ok(config)
+    pub agency_verkey: String,
+    pub remote_to_sdk_did: String,
+    pub remote_to_sdk_verkey: String,
+    pub sdk_to_remote_did: String,
+    pub sdk_to_remote_verkey: String,
```

Comment:
`AgencyClientConfig` may be more fitting? Not sure.

Reply:
renamed